src/lib/blueprint.test.ts

```diff
@@ -71,3 +76,251 @@ test('createProperties: uses provided values', (t) => {
     'isUndocumented should be true when x-undocumented is provided',
   )
 })
+
+test('Method detection for different endpoints', (t) => {
```

Comment:
Looks good, but prefer to make a test for each case, e.g.,

```
test('getPreferredMethod: post only', (t) => {
test('getPreferredMethod: get and post only', (t) => {
```

This is a lot easier to maintain and understand.